packages/@aws-cdk/aws-iam/lib/lazy-role.ts

```diff
@@ -108,21 +108,21 @@ export class LazyRole extends cdk.Resource implements IRole {
   /**
    * Grant the actions defined in actions to the identity Principal on this resource.
    */
-  public grant(identity: IPrincipal, ...actions: string[]): Grant {
+  public grant(identity: IGrantable, ...actions: string[]): Grant {
```

Comment:
Hmm. I don't think we can just change the input type in these functions. Based off some other changes we've seen lately, I think this might break python users, at the very least. I'm not positive about this, but we'll need to carefully consider how this would impact users. I think it's probably best to write new functions that use this as inputs instead or repurposing the old ones.

Reply:
If it occurs problems, it's out of scope of this PR, should be a separate issue.

I'll revert these arguments and change
```ts
role.grantAssumeRole(policy);
```
to
```ts
role.grantAssumeRole(policy.grantPrincilal);
```
and/or
```ts
Grant.addToPrincipal({ actions: [...], grantee: policy, resourceArgs: [...] });
```
as workaround in integ test.